providers/nats/nats.go

```diff
@@ -24,6 +25,11 @@ type Config struct {
 
 	// Prefix (optional).
 	Prefix string
+
+	// If true, keys will be split by delimiter "." into a nested map
+	// So, "a.b.c" results in {"a": {"b": {"c": "value" }}}
+	// Prefix will be included
+	Split bool
```

Comment:
Instead of `Split`, we can do `Delimiter string`, which allows the caller to pass any delimiter. That'd be consistent with the rest of koanf's semantics.

Reply:
Ah, yes, forgot about the fact that `.` is the default in NATS KV. We could just rename the field to `Unflatten bool` to be consistent with koanf's vocabulary then.